torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -42,6 +44,65 @@ inline int64_t getTimeUs() {
   return getTime() / 1000;
 #endif // USE_KINETO
 }
+}  // namespace
+
+namespace python_tracer {
+namespace {
+CallFn call_fn;
+TraceEventsFn get_events_fn;
+}  // namespace
+
+void registerFunctions(
+    CallFn call,
+    TraceEventsFn get_events) {
+  call_fn = call;
+  get_events_fn = get_events;
+}
+
+void call(Command c) {
```

Comment:
I just hate boilerplate, and having a bunch of:
```
CallFn blah_ptr

void blah() {
  if (blah_ptr != nullptr) {
    blah_ptr();
  }
}
```
stacked in a row was irritating me. (I care more about style than I probably should...)

Yeah, longer term I think the interface will get more complicated. We will want to support:
1) Dynamically adding threads. (E.g. a new dataloader thread is created.)
2) Reentrance. (Maybe?)
3) And indeed, probably multiple subscribers.

Right now I'm putting the tracing logic in `profiler_kineto.h / profiler_kineto.cpp` because it's expedient. (And the only thing that's using it.) Longer term I expect we'll need a C++ only header that goes in `libtorch` to control the tracer, and the C++ with Python.h one that goes in `python_libtorch`.

Reply:
yeah not a blocker and can live with this for now. its a change that can easily be added later